ochazuke/webhooks/__init__.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Flask Blueprint for our "webhooks" module.webhooks.
+
+See https://developer.github.com/webhooks/ for what is possible.
+"""
+
+import json
+import logging
+
+from flask import Blueprint
+from flask import request
+
+from ochazuke import app
+
+from .helpers import is_github_hook
+from .helpers import extract_issue_event_info
+from .helpers import update_db
+from .helpers import process_label_event_info
+from .helpers import process_milestone_event_info
+
+webhooks = Blueprint('webhooks', __name__, url_prefix='/webhooks')
+
+
+@webhooks.route('/issues', methods=['POST'])
+def issues_hooklistener():
+    """Listen for the "issues" webhook event.
+
+    By default, we return a 403 HTTP response.
+    """
+    if not is_github_hook(request):
+        return ('Move along, nothing to see here', 401, {'Content-Type':
+                                                         'text/plain'})
+    payload = json.loads(request.data)
+    event_type = request.headers.get('X-GitHub-Event')
+
+    # Treating issue events
+    if event_type == 'issues':
+        action = payload.get('action')
+        # We don't care about assignment events.
+        if action != ('assigned' or 'unassigned'):
+            # Extract relevant info to update issue and event tables.
+            issue_event_info = extract_issue_event_info(payload)
+            update_db(issue_event_info, action)
+            return ('Yay! Data! *munch, munch, munch*', 202,
+                    {'Content-Type': 'text/plain'})
```

Comment:
I was wondering this, too. GitHub keeps a log of recent webhook payloads, so they can be re-triggered manually. (I can't find the part of the docs that specifies how long or how many it keeps at a time.) That's probably not a solution, but it's a good thing to know in case of some future disastrous failure.

I would think we'd definitely want to log failures, and maybe set a certain number of re-tries before saving the data in a temp file or something? Not sure what all the options would be. I'll open an issue.